### PR TITLE
Allow for Symfony 6

### PR DESCRIPTION
**Pull request by tacman**
Symfony 6 will be released in a few days.  This PR allows for it, and fixes a few of the deprecation errors seen with the upcoming 5.4.

I'm not sure how to handle the phpstan errors, and setting up neon.  If someone can help set that up, I can help with fixing the deprecations for 5.4.

I made some changes to the build matrix, but I think it can be improved, in particular I'm not sure when with_imagine and with_oneupbundle should be used.

### CODE REVIEW

File: src/Adapter/AdapterInterface.php [hunk 1]
```diff
@@ -25,5 +25,5 @@ public function getObjectFromArgs($event);
      *
      * @param object $event The event
      */
-    public function recomputeChangeSet($event);
+    public function recomputeChangeSet($event): void;
```
Comment (garak) [inline src/Adapter/AdapterInterface.php:28]: we can't change interface signature without a BC break.
Same applies for adding type-hints to non-final classes

**Review by garak — CHANGES_REQUESTED** [review]

**Comment by tacman** [comment]
Oh.  Hmm.

So maybe I should drop this PR altogether.

Given the popularity of this bundle and the imminent release of Symfony 6,
can you or someone make the changes to support it?

Or I can continue to try.

**Comment by tacman** [comment]
Ah, okay!  Would you like to do that?  Or can we get the Symfony 6 support
in without fixing the phpstan errors?

**Comment by tacman** [comment]
Can you point me in the right direction on fixing these?


  1x: Since symfony/framework-bundle 5.3: Not setting the
"framework.session.storage_factory_id" configuration option is deprecated,
it will default to "session.storage.factory.native" and will replace the
"framework.session.storage_id" configuration option in version 6.0.
    1x in MappingDebugClassCommandTest::testNotUploadableClass from
Vich\UploaderBundle\Tests\Command

Legacy deprecation notices (2)

Other deprecation notices (24)

  8x: Since symfony/framework-bundle 5.3: The
"session.storage.factory.service" service is deprecated, use
"session.storage.factory.native", "session.storage.factory.php_bridge" or
"session.storage.factory.mock_file" instead.
    4x in UploadTest::testFileIsUploadedWithFileType from
Vich\UploaderBundle\Tests\Functional
    4x in UploadTest::testFileIsUploadedWithImageType from
Vich\UploaderBundle\Tests\Functional

  8x: Since symfony/framework-bundle 5.3: The "session.storage.mock_file"
service is deprecated, use "session.storage.factory.mock_file" instead.
    4x in UploadTest::testFileIsUploadedWithFileType from
Vich\UploaderBundle\Tests\Functional
    4x in UploadTest::testFileIsUploadedWithImageType from
Vich\UploaderBundle\Tests\Functional

  8x: Since symfony/framework-bundle 5.3: The
"session.storage.metadata_bag" service is deprecated, create your own
"session.storage.factory" instead.
    4x in UploadTest::testFileIsUploadedWithFileType from
Vich\UploaderBundle\Tests\Functional
    4x in UploadTest::testFileIsUploadedWithImageType from
Vich\UploaderBundle\Tests\Functional

**Comment by tacman** [comment]
Perfect, thanks!

What version of phpunit should we be using?  Can we safely bump to 9.5?  I
think we're running on 8.5, but locally I test with php8.  I am quite
confused about phpunit and such.

There are 19 skipped tests right now, and I'm not at all sure how to handle
those, as they're deep in the code related to FlySystem, the cache, etc.
Can I hand this off to someone with more experience?

**Comment by garak** [comment]
> Perfect, thanks! What version of phpunit should we be using? Can we safely bump to 9.5? I think we're running on 8.5, but locally I test with php8. I am quite confused about phpunit and such.

We're already on PHPUnit 9.5

> There are 19 skipped tests right now, and I'm not at all sure how to handle those, as they're deep in the code related to FlySystem, the cache, etc. Can I hand this off to someone with more experience?

The skipped tests depend on the environment, if you manage to get the needed dependencies in GitHub actions, they will automatically unskip

**Comment by jbelelieu** [comment]
Is there any progress on merging this?

**Comment by garak** [comment]
@tacman can you address the previous remarks, please?
We're waiting for this PR to release a new tag

**Comment by tacman** [comment]
> The skipped tests depend on the environment, if you manage to get the needed dependencies in GitHub actions, they will automatically unskip

Can someone help me out with this?  I'm stuck on what to do, just not familiar enough with github actions to know what to do.

**Comment by garak** [comment]
> > The skipped tests depend on the environment, if you manage to get the needed dependencies in GitHub actions, they will automatically unskip
> 
> Can someone help me out with this? I'm stuck on what to do, just not familiar enough with github actions to know what to do.

Just leave them as they are

**Comment by sxsws** [comment]
What is the problem holding this back?

**Comment by garak** [comment]
> What is the problem holding this back?

CI failures

**Comment by CoalaJoe** [comment]
LTM like there are too many deprecations triggered. Should we increase the num of allow deprecations for PHP 8.1?

**Comment by sxsws** [comment]
It looks like to me that @tacman has done the coding has now asked for help to get it through the tests and the one person able to help isn't?

**Comment by tacman** [comment]
I wish I understood what's not working and how to fix it. I've spent hours
trying, and am stuck.  I'm happy to take advice, learn, contribute, etc.
Thanks for your comment, I'm hoping it helps move the PR forward.

**Comment by garak** [comment]
@tacman the phpstan issues are pretty simple to solve: just look for mentioned ignored errors and remove them from the baseline file.
The deprecations in PHP 8.1 are maybe a bit trickier, but we only need to solve the direct ones.

**Comment by garak** [comment]
Closing in favour of #1250 
Thanks @tacman for your effort.

**Comment by sxsws** [comment]
Why did it have to be a separate PR? the only difference I see is the 'fixed tests', which is the help this PR needed to move it forwards and merge it.

Feels like a kick in face to @tacman

**Comment by garak** [comment]
Just because this PR was not ready, and the other one was.

**Comment by tobias-93** [comment]
@sxsws I never meant to insult anyone, I just felt @tacman was struggling and wanted to help.
As you can see in my PR, it took quite some iterations to get all tests up to speed and I wasn't even sure if I could get it all working anyway.
Putting my progress in this branch (and hence this PR) would have given a lot of unneccessary interference.
Only when I was kind of sure everything would work I submitted the PR.
Of course my thanks go to @tacman for his effort, I think he was quite close.

**Comment by tacman** [comment]
I'll accept that, and hope to learn from it!

All good, happy solstice and such.

Tac

**Pull request closed without merging**